guppylang/hugr_builder/hugr.py

```diff
@@ -180,9 +191,12 @@ def update_op(self) -> None:
         Feeds type information from the in- and out-ports to the operation class to
         update signature information. This function must be called before serialisation.
         """
-        in_types = [t.to_hugr() for t in self.in_port_types]
-        out_types = [t.to_hugr() for t in self.out_port_types]
-        self.op.insert_port_types(in_types, out_types)
+        if isinstance(self.op.root, ops.BaseOp) and not isinstance(
```

Comment:
The problem is not the call to `insert_port_types` but the call to `t.to_hugr()` since for these ops, `t` could be a parametrised `PolyFuncType` which can't be embedded into a type. I'll add a comment explaining what's going.

The proper way to deal with this issue would be adding a `Function` edge kind but this would have needed a big refactor of how Guppys internal Python Hugr data structure works. Since we're going to write a proper Python Hugr builder anyways, I thought it's probably not worth the effort.